### PR TITLE
Oop school library: changes

### PR DESCRIPTION
**Pull request by Paul-tes**
- Added a  class Person with the following:
   - Instance vars: @id, @name, and @age.
   - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have
   - default values of "Unknown" and true.
   - Getters for @id, @name, and @age.
   - Setters for @name and @age.
   - Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
   - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
- Added a class Student with the following:
   - Inherits from Person.
   - Constructor extends parent's constructor by adding @classroom and a parameter for it.
   - Method play_hooky that returns "¯\(ツ)/¯".
- Added a class Teacher with the following:
   - Inherits from Person.
   - Constructor extends parent's constructor by adding @specialization and a parameter for it.
   - Override can_use_services? so it always returns true.
- Each class are saved in a separate file.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, classroom, parent_permission: true, name: 'Unknown')
+    super(age, parent_permission, name)
+    @classroom = classroom
+  end
```
Comment (anagudelogu) [inline student.rb]: - [x] You did a great job working on this milestone! 💯 However, there's one catch in your code that will cause bugs when creating a Student. If you look closely at your Person class (super) initialize method, it expects two hash parameters: `parent_permission` and `name`. However, in your Student class, you're passing them as regular parameters, causing an argument error. Fixing this is very straightforward:

```suggestion
  def initialize(age, classroom, parent_permission: true, name: 'Unknown')
    super(age, parent_permission: parent_permission, name: name)
    @classroom = classroom
  end
```

You have to pass them as hashes! and that's it. 🤓 🔧

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, specialization, name: 'Unknown')
+    super(age, true, name)
+    @specialization = specialization
+  end
```
Comment (anagudelogu) [inline teacher.rb]: - [x] The same happens here, please consider fixing it! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @Paul-tes, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @Paul-tes      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯


- Good implementation of the required changes👍 
- clean code ✔ 👍
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**